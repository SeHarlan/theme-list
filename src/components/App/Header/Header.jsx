import React from 'react';
import PropTypes from 'prop-types';

const Header = () => (
  <header>
    <h1>Morty and Rick</h1>
    <em>Not Alive or Alive</em>

  </header>
);

Header.propTypes = {};

export default Header;
