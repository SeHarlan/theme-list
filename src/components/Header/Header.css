.Header {
  text-align: center;

  h1 {
    margin: 0;
    padding: 2rem;
  }
  label {
    margin: 0 auto;
    margin-top: 1rem;
  }
}