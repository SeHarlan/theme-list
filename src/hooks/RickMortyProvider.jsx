import React, { createContext, useState, useEffect, useContext } from 'react';
import PropTypes from 'prop-types';
import { fetchCharacters } from '../services/rickMorty';

const RickMortyContext = createContext();

export const RickMortyProvider = ({ children }) => {
  const [aliveCharacters, setAliveCharacters] = useState([]);
  const [deadCharacters, setDeadCharacters] = useState([]);
  const [page, setPage] = useState(1);
  const [totalPages, setTotalPages] = useState(25);
  const [theme, setTheme] = useState('Alive');

  const incPage = () => setPage(prev => (prev + 1));
  const decPage = () => setPage(prev => (prev - 1));

  useEffect(() => {
    fetchCharacters(page)
      .then(({ alive, dead, totalPages }) => {
        setAliveCharacters(alive);
        setDeadCharacters(dead);
        setTotalPages(totalPages);
      });
  }, []);

  return (
    <RickMortyContext.Provider value={{
      aliveCharacters, 
      deadCharacters, 
      theming: { theme, setTheme }, 
      paging: { decPage, incPage, page, totalPages }
    }}
    >
      {children}
    </RickMortyContext.Provider>
  );
};

RickMortyProvider.propTypes = { children: PropTypes.node };

export const useCharacters = () => {
  const { aliveCharacters, deadCharacters } = useContext(RickMortyContext);
  return { aliveCharacters, deadCharacters }; 
};

export const usePaging = () => {
  const { paging } = useContext(RickMortyContext);
  return paging;
};
export const useTheming = () => {
  const { theming } = useContext(RickMortyContext);
  return theming;
};
