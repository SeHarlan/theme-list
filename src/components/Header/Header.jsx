import React from 'react';
import Toggle from '../Toggle/Toggle';
import { useTheming } from '../../hooks/RickMortyProvider';

const MyHeader = () => {
  const { setTheme, theme } = useTheming();

  const toggle = () => {
    if(theme === 'Alive') setTheme('Dead');
    else setTheme('Alive');
  };

  return (
    <header>
      <h1>Morty and Rick</h1>
      <em>Not Alive or Alive</em>
      <Toggle action={toggle} checked={theme === 'Alive'} theme={theme}/>
    </header>
  );};


export default MyHeader;
