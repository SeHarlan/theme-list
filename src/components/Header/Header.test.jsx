import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import { RickMortyProvider } from '../../hooks/RickMortyProvider';

describe('Header component', () => {
  it('renders Header', () => {
    const wrapper = shallow(
      <RickMortyProvider>
        <Header />
      </RickMortyProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
