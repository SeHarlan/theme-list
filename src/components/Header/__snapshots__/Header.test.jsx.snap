// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<ContextProvider
  value={
    Object {
      "aliveCharacters": Array [],
      "deadCharacters": Array [],
      "paging": Object {
        "decPage": [Function],
        "incPage": [Function],
        "page": 1,
        "totalPages": 25,
      },
      "theming": Object {
        "setTheme": [Function],
        "theme": "Alive",
      },
    }
  }
>
  <MyHeader />
</ContextProvider>
`;
